src/main/flight/mixer.c

```diff
@@ -469,25 +485,49 @@ FAST_CODE_NOINLINE void mixTable(timeUs_t currentTimeUs)
         throttle = applyThrottleLimit(throttle);
     }
 
-    const bool airmodeEnabled = airmodeIsEnabled() || launchControlActive;
+    // use scaled throttle, without dynamic idle throttle offset, as the input to antigravity
+    pidUpdateAntiGravityThrottleFilter(throttle);
 
-#ifdef USE_YAW_SPIN_RECOVERY
-    // 50% throttle provides the maximum authority for yaw recovery when airmode is not active.
-    // When airmode is active the throttle setting doesn't impact recovery authority.
-    if (yawSpinDetected && !airmodeEnabled) {
-        throttle = 0.5f;   //
+#ifdef USE_DYN_LPF
+    // keep the changes to dynamic lowpass clean, without unnecessary dynamic changes
+    updateDynLpfCutoffs(currentTimeUs, throttle);
+#endif
+
+#ifdef USE_DYN_IDLE
+    // Apply idle up to throttle from dynamic idle, but after scaling, so that idle value doesn't change
+    float dynIdleThrottleOffset = (1 - throttle) * mixerRuntime.idleThrottleOffset;
+    if (mixerRuntime.dynIdleMinRps > 0.0f) {
+        throttle += dynIdleThrottleOffset;
     }
-#endif // USE_YAW_SPIN_RECOVERY
+#endif
 
-#ifdef USE_LAUNCH_CONTROL
-    // While launch control is active keep the throttle at minimum.
-    // Once the pilot triggers the launch throttle control will be reactivated.
-    if (launchControlActive) {
-        throttle = 0.0f;
+#ifdef USE_THRUST_LINEARIZATION
+    // counter compensate thrust linearization causing additional thrust at low throttle
+    // must be applied to the same throttle value that sets the motor output
+    throttle = pidCompensateThrustLinearization(throttle);
+#endif
+
+    // apply throttle boost when throttle moves quickly
+#if defined(USE_THROTTLE_BOOST)
+    if (throttleBoost > 0.0f) {
+        const float throttleHpf = throttle - pt1FilterApply(&throttleLpf, throttle);
+        throttle = constrainf(throttle + throttleBoost * throttleHpf, 0.0f, 1.0f);
+    }
+#endif
+
+    // send throttle value to blackbox, including scaling, throttle boost, thrust linear, but not dyn idle offset
+    mixerThrottle = throttle;
+#ifdef USE_DYN_IDLE
+    // adjust the idle offset out of the logged value
+    if (mixerRuntime.dynIdleMinRps > 0.0f) {
+        mixerThrottle -= dynIdleThrottleOffset;
```

Comment:
I can see what you are trying to do here, but I think the way you are doing it is wrong - since  throttle scaling, throttle boost, and thrust linear all act in a scaling way on the throttle value, the amount of dynamic idle offset that is part of the throttle value will be different at this point from what was added above, so it cannot simply be subtracted - worst case this will end up with a negative throttle value.
Maybe another way to do this will be to calculate the offset as a fraction of the throttle value when it is added above, and then remove the same fraction here?

Reply:
To be honest I couldn't figure a better way.  I wasn't happy with how I did it, but it worked properly as is.
There is a complex series of throttle manipulations going on here, and the outcome of later manipulations depends on what happened previously.
If there is a tidier way to achieve the same end result, then I'll happily implement it.

Reply:
> @ctzsnooze:
> 
> > removes transient throttle limit / AIRMODE_LPF code, since dynamic idle largely solves that problem, and it has been little used since added some time back, and is off by default
> 
> Can you please supply the data to back your claim of 'little used', to allow reviewers to form their own opinion of how much use it has had?

Not readily, unfortunately.  

It was recommended to not activate it at the same time as rpm filtering, so as rpm filtering becomes more ubiquitous, the scope for it's application becomes less and less.  

Perhaps users who would value retaining it can speak up here or during the beta testing phase.

Reply:
My proposal would be:

```
    float dynIdleThrottleOffset = (1 - throttle) * mixerRuntime.idleThrottleOffset;
    float throttleOffsetRatio = 0;
    if (mixerRuntime.dynIdleMinRps > 0.0f) {
        throttle += dynIdleThrottleOffset;
        throttleOffsetRatio = dynIdleThrottleOffset / throttle;
    }

[do throttle adjustments]

#ifdef USE_DYN_IDLE
    // adjust the idle offset out of the logged value
    if (mixerRuntime.dynIdleMinRps > 0.0f) {
        mixerThrottle *= (1 - throttleOffsetRatio);
    }
```

Still not ideal, as there is an additive component to the throttle adjustments as well.
But at least it avoids underruns.



> It was always a niche bit of code, I know very few people actually using it, and fewer actively benefitting from it.

I find the use of this kind of statement as a rationale for changes to be problematic - it is not known if your sample set is representative of the entirety of Betaflight users, so extrapolating from it for all users is not a sound approach.
If you postulate that this is the case, I think a good approach will be to collect statistics on usage in the first place, and then make a decision on removal of features once we have statistics.

Reply:
Yes I agree, usage statistics would be great.  However we have pressure on flash memory and we could alternately ask around to see how many people actively seem to use it.  I really think it would be OK to take it out at this point and see if people complain about doing that.  Its not hard to put back in if we must.  
The main argument for taking it out is the strong adoption of rpm filtering, and the need to not run both dynamic idle and transient throttle limit at the same time.

Reply:
Usage statistics in Betaflight are collected in configurator, and not in the firmware, so the argument about 'pressure on flash memory' is not applicable.
Again, my proposal is to leave this in, make it accessible in configurator and collect statistics on usage, and then make an informed decision on removal once this has happened.